src/Executor/DoctrineQueryBuilder/AutoJoin.php

```diff
@@ -83,10 +192,15 @@ private function analizeJoinedTables(QueryBuilder $queryBuilder)
     private function autoJoin(QueryBuilder $queryBuilder)
     {
         foreach ($this->expectedJoinChains as $tablesToJoin) {
+            // if the table is an embeddable, the property needs to be removed
+            if (array_search(implode('.', $tablesToJoin), $this->embeddables) !== false) {
+                array_pop($tablesToJoin);
+            }
+
             // check if the first dimension is a known alias
             if (isset($this->knownAliases[$tablesToJoin[0]])) {
                 $joinTo = $tablesToJoin[0];
-                array_pop($tablesToJoin);
+                array_shift($tablesToJoin);
```

Comment:
@K-Phoen I changed this because it looks like a bug to me. It was causing issues when performing complex joins (association of association). Please confirm that this change make sense.